skipruntime-ts/core/src/utils.ts

```diff
@@ -1,88 +1,63 @@
  */
-export class Sum implements Reducer<number, number> {
-  default = 0;
-
-  add(accum: number, value: number): number {
-    return accum + value;
-  }
-
-  remove(accum: number, value: number): Nullable<number> {
-    return accum - value;
-  }
+export class Sum implements NativeStub, Reducer<number, number> {
+  [sknative] = "sum";
+
+  // Lie to TypeScript that this implements Reducer, but leave out any implementations
+  // since we provide a native implementation.
+  default!: number;
+  add!: (accum: number) => number;
+  remove!: (accum: number) => Nullable<number>;
 }
 
 /**
  * `Reducer` to maintain the minimum of input values.
  *
  * A `Reducer` that maintains the minimum of values as they are added and removed from a collection.
  */
-export class Min implements Reducer<number, number> {
-  default = null;
-
-  add(accum: Nullable<number>, value: number): number {
-    return accum === null ? value : Math.min(accum, value);
-  }
-
-  remove(accum: number, value: number): Nullable<number> {
-    return value > accum ? accum : null;
-  }
+export class Min implements NativeStub, Reducer<number, number> {
+  [sknative] = "min";
+
+  // Lie to TypeScript that this implements Reducer, but leave out any implementations
+  // since we provide a native implementation.
+  default!: number;
+  add!: (accum: number) => number;
+  remove!: (accum: number) => Nullable<number>;
 }
 
 /**
  * `Reducer` to maintain the maximum of input values.
  *
  * A `Reducer` that maintains the maximum of values as they are added and removed from a collection.
  */
-export class Max implements Reducer<number, number> {
-  default = null;
-
-  add(accum: Nullable<number>, value: number): number {
-    return accum === null ? value : Math.max(accum, value);
-  }
-
-  remove(accum: number, value: number): Nullable<number> {
-    return value < accum ? accum : null;
-  }
+export class Max implements NativeStub, Reducer<number, number> {
+  [sknative] = "max";
+
+  // Lie to TypeScript that this implements Reducer, but leave out any implementations
+  // since we provide a native implementation.
+  default!: number;
+  add!: (accum: number) => number;
+  remove!: (accum: number) => Nullable<number>;
 }
 
 /**
  * `Reducer` to maintain the count of input values.
  *
  * A `Reducer` that maintains the number of values as they are added and removed from a collection.
  */
-export class Count<T extends Json> implements Reducer<T, number> {
-  default = 0;
-
-  add(accum: number): number {
-    return accum + 1;
-  }
-
-  remove(accum: number): Nullable<number> {
-    return accum - 1;
-  }
-}
-
-/**
- * `Mapper` to count input values.
- *
- * A `Mapper` that associates each key in the output with the number of values it is associated with in the input.
- *
- * @remarks
- * If there are many values associated to keys in a collection, and they are updated frequently, using the `Count` `Reducer` instead could be more efficient.
- */
-export class CountMapper<
```

Comment:
Note that this is used in www/docs/functions.md.

Reply:
I've got a note to use `Count` there instead (and add a unit test exercising that native reducer), top of my list for when I'm properly back to work next week.